code/modules/lighting/lighting_object.dm

```diff
@@ -75,21 +75,50 @@
 
 	var/max = max(cr.cache_mx, cg.cache_mx, cb.cache_mx, ca.cache_mx)
 
-	color  = list(
-		cr.cache_r, cr.cache_g, cr.cache_b, 0,
-		cg.cache_r, cg.cache_g, cg.cache_b, 0,
-		cb.cache_r, cb.cache_g, cb.cache_b, 0,
-		ca.cache_r, ca.cache_g, ca.cache_b, 0,
-		0, 0, 0, 1
-	)
+	var/rr = cr.cache_r
+	var/rg = cr.cache_g
+	var/rb = cr.cache_b
+
+	var/gr = cg.cache_r
+	var/gg = cg.cache_g
+	var/gb = cg.cache_b
+
+	var/br = cb.cache_r
+	var/bg = cb.cache_g
+	var/bb = cb.cache_b
+
+	var/ar = ca.cache_r
+	var/ag = ca.cache_g
+	var/ab = ca.cache_b
+
 	#if LIGHTING_SOFT_THRESHOLD != 0
-	luminosity = max > LIGHTING_SOFT_THRESHOLD
+	var/set_luminosity = max > LIGHTING_SOFT_THRESHOLD
 	#else
-	// Because of floating points™️, it won't even be a flat 0.
+	// Because of floating points�?, it won't even be a flat 0.
 	// This number is mostly arbitrary.
-	luminosity = max > 1e-6
+	var/set_luminosity = max > 1e-6
 	#endif
 
+	if((rr & gr & br & ar) && (rg + gg + bg + ag + rb + gb + bb + ab == 8))
```

Comment:
Because xg/xb follow xr in almost every situation this mixed check actually provides better performance than either addition or binary & checks alone.